clang-tools-extra/test/clang-tidy/checkers/readability/redundant-smartptr-get.cpp

```diff
@@ -20,6 +20,47 @@ struct shared_ptr {
   explicit operator bool() const noexcept;
 };
 
+template <typename T>
+struct vector {
+  vector();
+  bool operator==(const vector<T>& other) const;
+  bool operator!=(const vector<T>& other) const;
+  unsigned long size() const;
+  bool empty() const;
+
+  // Basic iterator implementation for testing
+  struct iterator {
```

Comment:
This boilerplate is excessive, you can just say the iterator is a pointer to T.
Also definitions aren't needed for all methods in the class, a declaration is more than enough

Reply:
modified

Reply:
Got it. Made the changes.